From: Eli Cohen <eli@mellanox.com>
Date: Sat, 4 Jun 2016 15:15:18 +0300
Subject: [PATCH] IB/core: Fix query port failure in RoCE
Patch-mainline: v4.7-rc5
Git-commit: d7012467a95b767b4d3beb2e027aa24a83f12f0f
References: bsc#966170 FATE#320225 bsc#966172 FATE#320226

Currently ib_query_port always attempts to to read the subnet prefix by
calling ib_query_gid(). For RoCE/iWARP there is no subnet manager and no
subnet prefix. Fix this by querying GID[0] only for IB networks.

Fixes: fad61ad4e755 ('IB/core: Add subnet prefix to port info')
Signed-off-by: Eli Cohen <eli@mellanox.com>
Signed-off-by: Leon Romanovsky <leon@kernel.org>
Reviewed-by: Steve Wise <swise@opengridcomputing.com>
Signed-off-by: Doug Ledford <dledford@redhat.com>
Acked-by: David Chang <dchang@suse.com>
---
 drivers/infiniband/core/device.c |    3 +++
 1 file changed, 3 insertions(+)

--- a/drivers/infiniband/core/device.c
+++ b/drivers/infiniband/core/device.c
@@ -680,6 +680,9 @@ int ib_query_port(struct ib_device *devi
 	if (err || port_attr->subnet_prefix)
 		return err;
 
+	if (rdma_port_get_link_layer(device, port_num) != IB_LINK_LAYER_INFINIBAND)
+		return 0;
+
 	err = ib_query_gid(device, port_num, 0, &gid, NULL);
 	if (err)
 		return err;
